### PR TITLE
Support Redshift Enhanced VPC Routing

### PR DESCRIPTION
**Pull request by ewbankkit**
Add `enhanced-vpc-routing` action to the Redshift resource to support [Amazon Redshift Enhanced VPC Routing](http://docs.aws.amazon.com/redshift/latest/mgmt/enhanced-vpc-routing.html).

### CODE REVIEW

**Comment by kapilt** [comment]
is this something we should be doing in custodian vs a user doing in provisioning?

**Comment by kapilt** [comment]
like is this something you would enable blanket across the fleet?

**Comment by kapilt** [comment]
nevermind after reading it does indeed sound like something we'd want in custodian

File: c7n/resources/redshift.py [hunk 1]
```diff
@@ -245,6 +245,38 @@ def process_cluster_snapshot(self, cluster):
             ClusterIdentifier=cluster['ClusterIdentifier'])
 
 
+@actions.register('enhanced-vpc-routing')
```
Comment (kapilt) [inline c7n/resources/redshift.py]: one minor preference for actions to be verbs, where as this is a noun (delta to goal state based systems versus imperative)... so enable-vpc-routing with the boolean as 'value' versus enabled since we're tending towards the later as a common vocabulary for switches that go true/false. albeit that's not a great nomenclature either.

File: tests/test_redshift.py [hunk 2]
```diff
@@ -182,6 +182,20 @@ def test_redshift_snapshot(self):
...
+            'name': 'redshift-enhanced-vpc-routing',
+            'resource': 'redshift',
+            'filters': [
+                {'type': 'value',
+                 'key': 'EnhancedVpcRouting',
+                 'value': True}],
+            'actions': [{'type': 'enhanced-vpc-routing', 'enabled': False}]},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
```
Comment (kapilt) [inline tests/test_redshift.py]: can you query fetch and assert the value is now enhanced to assert the policy had intended effect.

Comment (ewbankkit) [inline tests/test_redshift.py]: It seems the Redshift cluster goes into a **modifying** status for a while before the attribute change sticks. I could fake a `describe_clusters` response in the unit test but that's not gaining us anything 😄.

File: c7n/resources/redshift.py (hunk 1 above)
Comment (kapilt) [inline c7n/resources/redshift.py]: sounds good albeit a bit wordy, what if we just dropped the enhanced, and called it `enable-vpc-routing`

File: tests/test_redshift.py (hunk 2 above)
Comment (kapilt) [inline tests/test_redshift.py]: querying and asserting that modified status would suffice. else i could completely change the implementation to do nothing and the test still passes. you don't have to fake it, you can query post the policy execution for the actual. the same placebo session factory thats being passed to the load policy can also be used to construct a client for the tests to create or verify scenarios.

File: c7n/resources/redshift.py (hunk 1 above)
Comment (kapilt) [inline c7n/resources/redshift.py]: put another way, `enhanced` is a marketing term ;-)

**Comment by ewbankkit** [comment]
Rebased and made changes as suggested.

**Comment by kapilt** [comment]
thanks, fwiw, i've given up on rebase in pr and merge commits, vs squash and merge at the end. 

afaics this is still missing validation of the redshift modification  aka

```
self.assertEqual(cluster['PendingModified'], {"EnhancedVpcRouting": false})
```

else as is the action implementation could be made a no-op and the tests would still pass without that.

**Comment by ewbankkit** [comment]
Enhanced the unit test to verify that the correct cluster attribute is modified.

File: tests/test_redshift.py [hunk 3]
```diff
@@ -182,6 +182,30 @@ def test_redshift_snapshot(self):
...
+                 'value': True}],
+            'actions': [{'type': 'enable-vpc-routing', 'value': False}]},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        # Ensure that the cluster starts to modify EnhancedVpcRouting value.
+        response = client.describe_clusters(
+            ClusterIdentifier=resources[0]['ClusterIdentifier'])
+        cluster = response['Clusters'][0]
```
Comment (kapilt) [inline tests/test_redshift.py]: nitpicking, but technically you need a self.assertEqual(resources[0]['ClusterIdentifier'], clusters['ClusterIdentifier']) else this could be comparing apples and oranges if there are multiple clusters extant.

Comment (ewbankkit) [inline tests/test_redshift.py]: Yes sir 😄 .

**Comment by kapilt** [comment]
LGTM
